src/dune_engine/build_system.mli

```diff
@@ -88,6 +88,7 @@ module Error : sig
 
     module Map : Map.S with type key = t
 
+    val gen : unit -> t
```

Comment:
Do you need to make this function public? Can you just make `make` generate a new id? It's only used for the tests anyway.

Reply:
Good point